### PR TITLE
Added get method for vmc_vm_stats module and corresponding tests

### PR DESCRIPTION
**Pull request by shaikres**
Added get function for vmc_vm_stats module and corresponding tests
Added supported util functions

### CODE REVIEW

File: tests/unit/conftest.py [hunk 1]
```diff
@@ -20,6 +26,14 @@ def mock_access_token(mock_response):
         yield vmc_access_token
 
 
+@pytest.fixture
+def mock_vcenter_headers(mock_response):
+    with patch("saltext.vmware.utils.vmc_vcenter_request.get_headers") as vcenter_headers:
+        mock_response.json = mock.Mock("mock-header")
+        vcenter_headers.return_value = mock_response
+        yield vcenter_headers
```
Comment (waynew) [inline tests/unit/conftest.py:34]: **suggestion** this fixture is probably less desirable. Currently it's possible to remove both `get_api_session_id` and `get_headers` from `utils/vmc_vcenter_request.py` and all the unit tests pass.

A possible approach would be just to mock `requests.post` to return something like:

```
with patch("requests.post", autospec=True) as fake_request:
    fake_request.json.return_value = {'expected json': 'dictionary goes here'}
    yield
```

File: src/saltext/vmware/utils/vmc_vcenter_request.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    """
+    api_session_id = get_api_session_id(hostname, username, password)
+    return {
+        vmc_constants.CONTENT_TYPE: vmc_constants.APPLICATION_JSON,
+        vmc_constants.VMWARE_API_SESSION_ID: api_session_id,
+    }
+
+
+def call_api(
```
Comment (waynew) [inline src/saltext/vmware/utils/vmc_vcenter_request.py:58]: **suggestion** this appears to be a near/exact duplicate to `vmc_request`'s `call_api`. Is there a good reason not to have a common function for calling the API?

**Review by waynew — CHANGES_REQUESTED** [review]
the `vmc_vm_stats` part looks good, but utils & tests need some things addressed

File: src/saltext/vmware/utils/vmc_vcenter_request.py (hunk 2 above)
Comment (shaikres) [reply src/saltext/vmware/utils/vmc_vcenter_request.py:58]: Headers part is different for VMC and vcenter APIs.  In order to keep the code simple and understandable, we kept it as separate.

File: tests/unit/conftest.py (hunk 1 above)
Comment (shaikres) [reply tests/unit/conftest.py:34]: For vcenter APIs to be working properly, we need to pass api_session_id (based on vcenter_host, username and password) as header. Therefore, we should not remove `get_api_session_id` and `get_headers` from `utils/vmc_vcenter_request.py`

**Comment by shaikres** [comment]
Addressed the review comments replacing this PR with [https://github.com/saltstack/salt-ext-modules-vmware/pull/116](url)

**Pull request closed without merging**